src/@types/graphql.ts

```diff
@@ -0,0 +1,1192 @@
+export type Maybe<T> = T | null;
+export type InputMaybe<T> = Maybe<T>;
+export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
+export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
+export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
+export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
+export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
+/** All built-in and custom scalars, mapped to their actual values */
+export type Scalars = {
+  ID: { input: string; output: string; }
+  String: { input: string; output: string; }
+  Boolean: { input: boolean; output: boolean; }
+  Int: { input: number; output: number; }
+  Float: { input: number; output: number; }
+  Date: { input: any; output: any; }
+  JSONObject: { input: any; output: any; }
+};
+
+export type AutomationAction = {
+  __typename?: 'AutomationAction';
+  alertRecipients?: Maybe<Array<Maybe<Scalars['String']['output']>>>;
```

Comment:
If these fields are nullable, then why do we need this `Maybe` type?

Reply:
That's a TS representation of how it's currently typed here: https://github.com/tnc-ca-geo/animl-api/blob/cb48e5dc149d01643b60c038837204d256180cce/src/api/type-defs/objects/AutomationRule.ts#L7-L13

This means that all of these are valid `AutomationAction` payloads:


```json5
{
  // ...
  alertRecipients: ["someRecipient@mail.com"]
}
```

```json5
{
  // ...
  alertRecipients: undefined
}
```


```json5
{
  // ...
  alertRecipients: []
}
```

```json5
{
  // ...
  alertRecipients: null
}
```

```json5
{
  // ...
  alertRecipients: [null]
}
```

Personally, I doubt that some of these are actually valid (particularly the last one).

The type should probably be specified as:

```gql
   type AutomationAction { 
     type: String! 
     alertRecipients: [String!]! 
     mlModel: String 
     confThreshold: Float 
     categoryConfig: JSONObject 
   } 
```

However I'm trying to avoid critiquing types unless they cause functional issues within the codebase's type completeness.

Reply:
In #197, #198, and #199 I had to update the types to make TS happy.

Reply:
Gotcha

Reply:
@alukach I think your policy of trying to avoid modifying the graphql types unless they cause issues is a good one for now. I have no doubt the graphql types can and should be tightened up but it could easily turn into a can of worms that produces some unexpected breaking issues. 

For example, in this case, an `AutomationAction` is used to provide details on how to execute an [Automation Rule](https://docs.animl.camera/fundamentals/automation-rules) for both `type: "run-inference"` and `type: "send-alert"` actions, and some of those props are only relevant to sending alerts (`alertRecipients`) while some are only relevant to requesting inference (`mlModel`, `confThreshold`, `categoryConfig`). So `alertRecipients` should not be required for inference-request-related `AutomationActions`, but perhaps if it's present the array should always have stings? So the type-def should be something like `alertRecipients: [String!]`?

For now, don't worry about this though. Not requesting a change just adding some context.